tools/CAT/src/pages/EditPage/index.tsx

```diff
@@ -1,95 +1,77 @@
 import { useEffect } from 'react';
-import EditPageDefault from './EditPageDefault';
-import EditPageWithTitle from './EditPageWithTitle';
+import EditPageDefault from './editPageDefault';
+import EditPageWithTitle from './editPageWithTitle';
 import Edit from '@/components/Edit';
-import Dialog from '@/components/Dialog';
+import { Link } from 'react-router-dom';
 import UploadFileWrapper from '@/components/UploadFileWrapper';
 import SpinnerWrapper from '@/components/Spinner';
 import { Button } from '@chakra-ui/react';
-import { useUpdatePageMutation } from '@/store/apis/categories';
-import { useGetPageQuery } from '@/store/apis/categories';
 import DeleteButton from '@/components/DeleteButton';
-import { useDeletePageMutation } from '@/store/apis/categories';
 import { useInitialPageState } from '@/initialStates/PageState';
+import { useParams } from 'react-router-dom';
+import { deletePage, updatePage } from '@/app/features/pages/pageThunks';
+import { useAppDispatch, useAppSelector } from '@/app/hooks';
+import { getPageData, getPageLoading } from '@/app/features/pages/pageSlice';
+import { fetchPageByPath } from '@/app/features/pages/pageThunks';
+import { useNavigate } from 'react-router-dom';
 
-interface EditPageProps {
-	handleClose: () => void;
-	parentUrl: string;
-	id: string;
-	subCategoryId: string;
-}
+export default function EditPage() {
+	const { category, subcategory, page } = useParams();
+	const viewUrl = `${
+		import.meta.env.VITE_API_URL
+	}/${category}/${subcategory}`;
+
+	const dispatch = useAppDispatch();
+	const pageData = useAppSelector(getPageData);
+	const isLoading = useAppSelector(getPageLoading);
+	const { pageDataState, setPageDataState } = useInitialPageState({});
+	const navigate = useNavigate();
 
-export default function EditPage({
-	handleClose,
-	parentUrl,
-	id,
-	subCategoryId,
-}: EditPageProps) {
-	const { pageData, setPageData } = useInitialPageState({
-		id,
-		subCategoryId,
-	});
-	const [deletePage] = useDeletePageMutation();
-	const { data, isLoading } = useGetPageQuery(id);
+	useEffect(() => {
+		dispatch(
+			fetchPageByPath(
+				`${category ? `${category}/` : ''}${
```

Comment:
I think you can write it like this:
const pageByPathUrl = [category, subCategory, page].filter(Boolean).join('/');
I think it looks better, but not sure if more clear @maryanchuk04 what do you think?

Reply:
BTW I think we do not expect have category or subcategory null here. So I think we can use simple join.